pkg/k8s/scanner/scanner_test.go

```diff
@@ -0,0 +1,213 @@
+											},
+										},
+									},
+								},
+								Properties: map[string]string{
+									"PkgID":   "k8s.gcr.io/kube-apiserver:1.21.1",
+									"PkgType": "oci",
+								},
+							},
+						},
+					},
+					{
+						Type: cdx.ComponentTypeContainer,
+						Name: "kind-control-plane",
+						Properties: map[string]string{
+							"architecture":     "arm64",
+							"host_name":        "kind-control-plane",
+							"kernel_version":   "6.2.15-300.fc38.aarch64",
+							"node_role":        "master",
+							"operating_system": "linux",
+						},
+						Components: []*core.Component{
+							{
+								Type:    cdx.ComponentTypeOS,
+								Name:    "ubuntu",
+								Version: "21.04",
+								Properties: map[string]string{
+									"Class": "os-pkgs",
+									"Type":  "ubuntu",
+								},
+							},
+							{
+								Type: cdx.ComponentTypeApplication,
+								Name: "node-core-components",
+								Properties: map[string]string{
+									"Class": "lang-pkgs",
+									"Type":  "golang",
+								},
+								Components: []*core.Component{
+									{
+										Type:    cdx.ComponentTypeLibrary,
+										Name:    "kubelet",
+										Version: "1.21.1",
+										Properties: map[string]string{
+											"PkgType": "golang",
+										},
+										PackageURL: &purl.PackageURL{
+											PackageURL: packageurl.PackageURL{
+												Type:       "golang",
+												Name:       "kubelet",
+												Version:    "1.21.1",
+												Qualifiers: packageurl.Qualifiers{},
+											},
+										},
+									},
+									{
+										Type:    cdx.ComponentTypeLibrary,
+										Name:    "containerd",
+										Version: "1.5.2",
+										Properties: map[string]string{
+											"PkgType": "golang",
+										},
+										PackageURL: &purl.PackageURL{
+											PackageURL: packageurl.PackageURL{
+												Type:       "golang",
+												Name:       "containerd",
+												Version:    "1.5.2",
+												Qualifiers: packageurl.Qualifiers{},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.TODO()
+			runner, err := cmd.NewRunner(ctx, flagOpts)
+			assert.NoError(t, err)
+			scanner := NewScanner(tt.clusterName, runner, flagOpts)
+			got, err := scanner.Scan(ctx, tt.artifacts)
+			sortNodeComponents(got.RootComponent)
+			sortNodeComponents(tt.want)
+			assert.Equal(t, tt.want, got.RootComponent)
+		})
+	}
+}
+
+type coreComponents []*core.Component
+
+func (a coreComponents) Len() int           { return len(a) }
+func (a coreComponents) Less(i, j int) bool { return a[i].Name < a[j].Name }
+func (a coreComponents) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+
+func sortNodeComponents(component *core.Component) {
+	nodeComp := findComponentByName(component, "node-core-components")
+	sort.Sort(coreComponents(nodeComp.Components))
```

Comment:
If you just want to sort by name, you don't need the new type implementing `sort.Interface`.

```suggestion
	sort.Slice(nodeComp.Components, func(i, j int) bool {
		return nodeComp.Components[i].Name < nodeComp.Components[j].Name
	})
```